src/empty.js

```diff
@@ -1,21 +1,36 @@
 var _curry1 = require('./internal/_curry1');
-var _hasMethod = require('./internal/_hasMethod');
 
 
 /**
- * `empty` returns an empty list for any argument, except when the argument satisfies the
- * Fantasy-land Monoid spec. In that case, this function will return the result of invoking
- * `empty` on that Monoid.
+ * Returns the empty value of its argument's type. The argument must be a
+ * [Monoid](https://github.com/fantasyland/fantasy-land#monoid), Array,
+ * Object, or String.
```

Comment:
At the moment the behaviour is left unspecified (returning `undefined` happens to fall out of the implementation). Throwing is another option. My preference is to throw given `null` or `/xxx/` or a value of any other unsupported type, but Ramda tends to follow JavaScript's lead by failing silently (`R.head([])` does not throw, for example).

If we agree that we should return `undefined` given a value of an unsupported type, I'll update the docstring accordingly.